Sources/xcodeproj/Project/XcodeProj.swift

```diff
@@ -27,8 +27,8 @@ public final class XcodeProj: Equatable {
             if pbxprojPaths.count == 0 {
                 throw XCodeProjError.pbxprojNotFound(path: path)
             }
-            let pbxProjData = try Data(contentsOf: pbxprojPaths.first!.url)
-            let context = ProjectDecodingContext()
+            let (pbxProjData, jsonDictionary) = try XcodeProj.readPBXProj(url: pbxprojPaths.first!.url)
+            let context = ProjectDecodingContext(jsonDictionary: jsonDictionary)
```

Comment:
I think this one can be optimised as well by passing a reference instead of a value type. If I'm not wrong, Swift creates a copy of the whole `jsonDictionary` when initialising the `ProjectDecodingContext` object.

Reply:
Nitpick: I'd also rename the name of the argument to `pbxProjDictionary`

Reply:
Fixed